### PR TITLE
feat(dependency): Migrated to go modules

### PR DESCRIPTION
**Pull request by harshthakur9030**
**Why is this PR required? What issue does it fix?**:
Go modules is a better dependency management than dep.

**What this PR does?**:
Migrates to use Go modules

**Does this PR require any upgrade changes?**:
 Goinkgo and Gomega were upgraded

**If the changes in this PR are manually verified, list down the scenarios covered:**:
Works on go 1.13 and 1.14

**Checklist:**
- [ ] Fixes #<issue number>
- [x] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Has the change log section been updated? 
- [ ] Commit has unit tests
- [ ] Commit has integration tests
- [ ] (Optional) Are upgrade changes included in this PR? If not, mention the issue/PR to track: 
- [ ] (Optional) If documentation changes are required, which issue on https://github.com/openebs/openebs-docs is used to track them:

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -162,21 +160,14 @@ vet:
...
 # shellcheck target for checking shell scripts linting
 .PHONY: shellcheck
 shellcheck: getshellcheck
 	find . -type f -name "*.sh" | grep -v "./vendor/*" | xargs /tmp/shellcheck-stable/shellcheck
 
 .PHONY: getshellcheck
 getshellcheck:
-	wget -c 'https://github.com/koalaman/shellcheck/releases/download/stable/shellcheck-stable.linux.x86_64.tar.xz' --no-check-certificate -O - | tar -xvJ -C /tmp/
-
+ 	wget -c 'https://github.com/koalaman/shellcheck/releases/download/stable/shellcheck-stable.linux.x86_64.tar.xz' --no-check-certificate -O - | tar -xvJ -C /tmp/
```
Comment (akhilerm) [inline Makefile:170]: This is not required right? The change is already present.

Comment (harshthakur9030) [reply Makefile:170]: The previous build failed for shell check, I assumed I forgot to add the URL fix but even now it fails. I can git reset and hard push once we figure out what's causing it

Comment (akhilerm) [reply Makefile:170]: The error says not able to find the shellcheck binary in the path

Comment (harshthakur9030) [reply Makefile:170]: It finds it in other architectures.

Comment (akhilerm) [reply Makefile:170]: shellcheck is enabled only in `amd64`, thats why its not failing in other architectures.

Comment (harshthakur9030) [reply Makefile:170]: Oh. How do we fix it?

**Comment by akhilerm** [comment]
The changes looks good, Tried building and it all works fine. Can you also check if we need to pass `-mod` argument to `gox` command while building?

File: Makefile [hunk 2]
```diff
@@ -251,7 +244,8 @@ docker.exporter: build.exporter Dockerfile.exporter
 .PHONY: deps
 deps: header
 	@echo '--> Resolving dependencies...'
-	dep ensure
+	go mod vendor
+	go mod tidy
```
Comment (mynktl) [inline Makefile:248]: sequence can be wrong. tidy updates the module list and vendor will make copy of dependency. Better to run `go mod tidy` first then `go mod vendor`

Comment (mynktl) [reply Makefile:248]: You may need to check go.mod/go.sum to verify dependency. Check  [this example](https://github.com/openebs/velero-plugin/blob/master/Makefile#L90)  for verification.

**Comment by mynktl** [comment]
@harshthakur9030 You also need to remove `Gopkg.lock` and `Gopkg.toml` file since these files are not required anymore.

**Comment by harshthakur9030** [comment]
Done! @mynktl

**Comment by harshthakur9030** [comment]
I just noticed the integration test failed again.

**Comment by harshthakur9030** [comment]
Before merging, we might need to fix the weird error for integration test. No clue why that's happening @akhilerm @mynktl

**Comment by akhilerm** [comment]
> Before merging, we might need to fix the weird error for integration test. No clue why that's happening @akhilerm @mynktl

I am trying to reproduce it locally, But with no result.

**Comment by akhilerm** [comment]
@harshthakur9030 , if you look at the log messages [here](https://travis-ci.org/github/openebs/node-disk-manager/jobs/694892781#L1092), you can see that while running the integration test. the modules are not fetched. When tried locally, first it fetched the modules and then ran the integration test. Not sure if it is the exact cause. you may need to check on it.

**Comment by harshthakur9030** [comment]
Oh, can you compare your go env with Travis?

**Comment by harshthakur9030** [comment]
Closing this. I will send a new PR with clean commit history

**Pull request closed without merging**